extension/src/NABfunctions.ts

```diff
@@ -411,3 +411,47 @@ export async function updateAllXlfFiles() {
 
     console.log('Done: Update all XLF files');
 }
+
+export async function createNewTargetXlf() {
+    console.log("Running: createNewTargetXlf");
+    const targetLanguage: string | undefined = await getUserInput({ placeHolder: "Language code e.g sv-SE" });
+    const selectedMatchBaseApp = await getQuickPickResult(["Yes", "No"], { canPickMany: false, placeHolder: "Match translations from BaseApp?" });
+
+    if (isNullOrUndefined(targetLanguage) || targetLanguage.length === 0) {
+        throw new Error("No target language was set.");
+    }
+    try {
+        const gXlfFile = await WorkspaceFunctions.getGXlfFile();
+        const translationFolderPath = WorkspaceFunctions.getTranslationFolderPath();
+        const matchBaseAppTranslation: boolean = (selectedMatchBaseApp === "Yes");
+        const targetXlfFilepath = path.join(translationFolderPath, `${targetLanguage}.xlf`);
+        if (fs.existsSync(targetXlfFilepath)) {
+            throw new Error(`File already exists: '${targetXlfFilepath}'`);
+        }
+
+        console.log(`Creating new target xlf for language: ${targetLanguage}.\nMatch translations from BaseApp: ${matchBaseAppTranslation}.\nSaving file to path: ${targetXlfFilepath}`);
+        const targetXlfDoc = Xliff.fromFileSync(gXlfFile.fsPath);
+        targetXlfDoc.targetLanguage = targetLanguage;
+        if (matchBaseAppTranslation) {
+            let numberOfMatches = await LanguageFunctions.matchTranslationsFromBaseApp(targetXlfDoc);
+            vscode.window.showInformationMessage(`Added ${numberOfMatches} suggestions from Base Application in ${targetXlfFilepath.replace(/^.*[\\\/]/, '')}.`);
```

Comment:
why not have the filename in it's own variabel above and use that instead of regex? ;)

Reply:
Hehehe copy paste ;)